pennylane/templates/subroutines/flip_sign.py

```diff
@@ -0,0 +1,185 @@
+
+    .. seealso:: :func:`~.relevant_func`, :class:`~.RelevantClass` (optional)
+
+    .. details::
+
+        :title: Usage Details
+
+        The template is used inside a qnode.
+        The number of shots has to be explicitly set on the device when using sample-based measurements:
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit", wires=4, shots = 1)
+
+            @qml.qnode(dev)
+            def circuit():
+               for wire in list(range(4)):
+                    qml.Hadamard(wires = wire)
+               qml.FlipSign([1,0,0,0], wires = list(range(4)))
+               return qml.sample()
+
+            circuit()
+
+        The result for the above circuit is:
+
+            >>> print(circuit.draw())
+            0: ──H─╭FlipSign───┤  Sample
+            1: ──H─├FlipSign───┤  Sample
+            2: ──H─├FlipSign───┤  Sample
+            3: ──H─╰FlipSign───┤  Sample
+
+    """
+
+    num_wires = AnyWires
+
+    def __init__(self, n, wires, do_queue=True, id=None):
+
+        if type(wires) is not int and len(wires) == 0:
+            raise ValueError("expected at least one wire representing the qubit ")
+
+        if type(n) is int:
+            if n >= 0:
+                n = self.to_list(n, len(wires))
+            else:
+                raise ValueError(
+                    "expected an integer equal or greater than zero for basic flipping state"
+                )
+        else:
+
+            if self.is_list_typeof(n, int):
+                n = self.to_number(n)
+            else:
+                raise ValueError(
+                    "expected an integer binary array or integer number for basic flipping state "
+                )
+
+        self._hyperparameters = {"arr_bin": n}
+        super().__init__(wires=wires, do_queue=do_queue, id=id)
+
+    @staticmethod
+    def is_list_typeof(arr, type_val):
+        r"""Check if array is for a given type or not
+        Args:
+            type_val (type): Type to infer and check for array
+            arr (array[object]): Integer binary array with regarding basis state
+
+        Returns:
+            (bool): boolean that checks whether array is binary or not
+        """
+        return np.array_equal(arr, np.array(arr).astype(type_val))
+
+    @staticmethod
+    def to_list(n, n_wires):
+        r"""Convert an integer into a binary integer list
+        Args:
+            n (int): Basis state as integer
+            n_wires (int): Numer of wires to transform the basis state
+
+        Raises:
+            ValueError: "cannot encode n with n wires "
+
+        Returns:
+            (array[int]): integer binary array
+        """
+        if n >= 2**n_wires:
+            raise ValueError(f"cannot encode {n} with {n_wires} wires ")
+
+        b_str = f"{n:b}".zfill(n_wires)
+        bin_list = [int(i) for i in b_str]
+        return bin_list
+
+    @staticmethod
+    def to_number(arr_bin):
```

Comment:
If I am not mistaken, this function is not necessary anymore if you follow the suggestions of the __init__ part😄

Reply:
you are right again! delete!